### DISCUSSION TITLE
What methods and attributes does FastAPI actually use in BaseModel schemas?

### DISCUSSION THREAD

**Discussion by bjpietrzak** [fastapi/fastapi#10963] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List, Optional, Dict
from pydantic import BaseModel


class ProcessedReview(BaseModel):
    review: str
    thumbs_up_count: int
    sentiment: int
```

### Description

What pydantic BaseModel methods and attributes are used by FastAPI in validation of the request and response data?

### Operating System

macOS

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

3.11

**Comment by YuriiMotov** [comment]
You can get this information by inspecting the source codes
